jcl/src/java.base/share/classes/openj9/internal/criu/InternalCRIUSupport.java

```diff
@@ -940,16 +950,27 @@ private static void setRestoreJavaProperties() {
 	}
 
 	private static void clearInetAddressCache() {
-		Field jniaa = AccessController.doPrivileged((PrivilegedAction<Field>) () -> {
-			Field jniaaTmp = null;
+		Field jniaa;
+		/*[IF JAVA_SPEC_VERSION >= 24]*/
+		try {
+			jniaa = SharedSecrets.class.getDeclaredField("javaNetInetAddressAccess"); //$NON-NLS-1$
+			jniaa.setAccessible(true);
+		} catch (NoSuchFieldException e) {
+			// ignore exceptions
+		}
+		/*[ELSE] JAVA_SPEC_VERSION >= 24 */
+		jniaa = AccessController.doPrivileged((PrivilegedAction<Field>) () -> {
+			Field jniaaTmp;
 			try {
 				jniaaTmp = SharedSecrets.class.getDeclaredField("javaNetInetAddressAccess"); //$NON-NLS-1$
 				jniaaTmp.setAccessible(true);
 			} catch (NoSuchFieldException | SecurityException e) {
 				// ignore exceptions
+				jniaaTmp = null;
```

Comment:
This is redundant. The suggestion was to put this in the Java24+ catch block (after line 959), without which, Java 24+ will not compile.